rfcs/0004-pablo-distribution.adoc

```diff
@@ -0,0 +1,480 @@
+
+= l_r / (sum l_r) * sum a_n
+
+"With this adjusted value all later additions to LP shares have been negated when receiving fees for earlier LPs."
+
+
+++++
+
+[#_pblo_staker_trading_fee_distribution]
+==== PBLO Staker Trading Fee Distribution
+
+This is the reward a `PBLO` staker receives from the trading fees of Pablo pools. It is equal to the protocol fee charged on Pablo pools. This can be accomplished by calling the already existing `StakingReward.transfer_reward` interface as follows. According to product there is also a need to convert whatever the fee asset in to PBLO to create a demand/additional value for PBLO.
+
+[plantuml,images/pablo-fNFT-pblo-staking-fee-distro,png]
+----
+start
+->after Pablo swap,
+fee=x,
+fee token=QUOTE,
+from = transaction_origin;
+:swap x of QUOTE to y of PBLO;
+note left: how to avoid fee\nswap recursion? \n I wonder if we really need this?
+:StakingReward::transfer_reward(
+asset=PBLO,
+reward_asset: PBLO
+from=transaction_origin,
+amount=y,
+keep_alive=false
+);
+stop
+----
+
+Will it need a change in https://github.com/ComposableFi/composable/blob/main/frame/composable-traits/src/staking_rewards.rs#L96[this] ?
+
+=== Pallet Staking Rewards - LP/PICA/PBLO/Other Token Staking Reward Pools
+
+This section covers how the staking rewards are distributed using the https://github.com/ComposableFi/composable/tree/main/frame/staking-rewards[staking rewards pallet].
+
+==== Analysis of Reward Calculations
+
+In order to create the necessary reward pool as well as the rewarding rate for stakers the following model can be used. It tries to address the following constraints
+
+. Allow addition of new stakers at anytime to a pool, start earning immediate rewards
+. Allow more realtime calculation of rewards on-demand for a given pool for a given user.
+. Allow shorter reward pool calculation epoch with the use of the reward rate.
+. Allow specification of reward rate for a pool (even setting a dynamically changing rate)
+. Allow expansion of rewards pools realtime.
+
+[stem]
+++++
+"Let's define the following terms for a given staking reward pool,"
+
+"Pre-defined reward rate (say per second)" = r
+
+"Pre-defined reward calculation epoch in seconds" = t
+
+"Reward per calculation epoch" = r.t
+
+"Previous total reward pool before the current epoch" = P
+
+"Assuming there is a per epoch calculation which adds to the pool,"
+"the total reward pool for the current epoc,"
+
+P_(current) = P + rt
+
+"Reward pool shares for n stakers"
+
+= sum (s_n)
+
+"Where " n_(th) " staker share is " s_n
+
+"Existing " n_(th) " staker reward,"
+
+x _n = P_(current) . s_n / (sum s_n)
+
+------------------------------------------------
+
+"When adding a new staker n+1, existing stakers(n) reward would be,"
+
+x_n = P_(current) . s_n / (sum s_(n+1))
+
+"As this is less than what is expected above, an adjustment " delta P " to total reward pool can be made to allow realtime reward calculations,"
```

Comment:
i doubt it will work. so you may try. may be i do not know.

Reply:
For the cases we are considering above (5 points), this works as the math suggests here. But there may be cases which I have not considered in relation to compounding, extending etc., that I may need your help in resolving.